### PR TITLE
GCP Identity Provider

### PR DESCRIPTION
**Pull request by jpcoenen**
Adds a new credential type for GCP Service Account, which allows a service running on GCP to use _keyless_ authentication.

Introduces a new (currently hidden) `secrethub service gcp` subcommand and allows `gcp` as an Identity Provider.

Please note: this feature starts in private beta. By default, it will not be enabled on your namespace. If you want to give it a try, reach out to us at support@secrethub.io.

### CODE REVIEW

File: internals/secrethub/service_gcp_init.go [hunk 1]
```diff
@@ -0,0 +1,302 @@
...
+		cmd.serviceAccountEmail = strings.TrimSpace(serviceAccountEmail)
+	}
+
+	if cmd.kmsKeyResourceID == "" {
+		kmsKey, err := ui.AskAndValidate(cmd.io, "What is the Resource ID of the KMS-key that should be used for encrypting the service's account key?\n", 3, api.ValidateGCPKMSKeyResourceID)
+		if err != nil {
+			return err
+		}
+		cmd.kmsKeyResourceID = strings.TrimSpace(kmsKey)
+	}
```
[the remark this answers is not in the document]
Comment (jpcoenen) [reply internals/secrethub/service_gcp_init.go:105]: Fair point. The reason is that I left it this way, is that otherwise, it would get fairly complex while I suspect there is no real use-case for this.

The reason for it getting more complex, is that, to make it feel right, you'd have to extract the project id from the given flag and use that for the choice.

At the same time, if you have access to the exact value of 1 of 2 parameters (because you provided is as a flag), you probably also have access to the other one. So copying it should not really be a problem.

In conclusion, I'd go with the simpler solution that covers practically all use-cases and therefore leave it unchanged. What do you say?

[the remark this answers is not in the document]
Comment (SimonBarendse) [reply internals/secrethub/service_gcp_init.go:105]: Ah I see. So the main difference between the two scenarios is that if neither parameters is supplied, we find it okay to prompt for the project ID, while if either of the parameters is supplied, we don't. Seems like a good reason to handle them separately :+1:

About extracting the project ID from the given flag: I agree we can leave this out for now. If it turns out supplying one value is used a lot, we can always add the options to the prompt later.
